packages/test-base/src/nativeDarwin/kotlin/io/realm/kotlin/test/platform/PlatformUtils.kt

```diff
@@ -35,7 +36,8 @@ import kotlin.time.Duration
 actual object PlatformUtils {
     actual fun createTempDir(prefix: String, readOnly: Boolean): String {
         // X is a special char which will be replace by mkdtemp template
-        val mask = prefix.replace('X', 'Z', ignoreCase = true)
+        val suffix = "-${Utils.createRandomString(4)}"
```

Comment:
Any reason not to just add an `xxxxxx`-suffix (after replacing X->Z) and rely on `mkdtemp`?

Reply:
I wasn't able to make `mkdtemp` work. It returns a pointer to the new path, but I cannot retrieve the actual data.

I have moved the implementation to use `mkdir` with our template.